### PR TITLE
Trim down framework references for Blazor dev server

### PR DESCRIPTION
**Pull request by captainsafia**
While investigating https://github.com/dotnet/aspnetcore/issues/30917, we realized that things seems to work when the `runtimeconfig.json` for the Blazor dev server doesn't contain a framework reference to `Microsoft.NETCore.App`. 

Generated runtimeconfig now looks like this:

Screen Shot 2021-03-31 at 2 15 18 PM

This is a stopgap solution.

### CODE REVIEW

File: src/Components/WebAssembly/DevServer/src/Microsoft.AspNetCore.Components.WebAssembly.DevServer.csproj [hunk 1]
```diff
@@ -11,6 +11,7 @@
     <GenerateDocumentationFile>true</GenerateDocumentationFile>
     <!-- Set this to false because assemblies should not reference this assembly directly, (except for tests, of course). -->
     <IsProjectReferenceProvider>false</IsProjectReferenceProvider>
+    <GenerateRuntimeConfigurationFiles>false</GenerateRuntimeConfigurationFiles>
```
Comment (HaoK) [inline src/Components/WebAssembly/DevServer/src/Microsoft.AspNetCore.Components.WebAssembly.DevServer.csproj:14]: Maybe add a comment here for clarity/posterity?

File: src/Components/WebAssembly/DevServer/src/blazor-devserver.runtimeconfig.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{
+  "runtimeOptions": {
+    "tfm": "net6.0",
+    "framework": {
+      "name": "Microsoft.AspNetCore.App",
+      "version": "6.0.0-preview.3.21167.1"
```
Comment (HaoK) [inline src/Components/WebAssembly/DevServer/src/blazor-devserver.runtimeconfig.json:6]: Do we need to worry about this version baked in?

Comment (pranavkm) [reply src/Components/WebAssembly/DevServer/src/blazor-devserver.runtimeconfig.json:6]: Could we do something like this? 
https://github.com/dotnet/aspnetcore/blob/main/src/Components/WebAssembly/BlazorManifest/src/BlazorManifest.csproj#L25-L28

File: src/Components/WebAssembly/DevServer/src/Microsoft.AspNetCore.Components.WebAssembly.DevServer.csproj (hunk 1 above)
Comment (captainsafia) [reply src/Components/WebAssembly/DevServer/src/Microsoft.AspNetCore.Components.WebAssembly.DevServer.csproj:14]: Good call!

File: src/Components/WebAssembly/DevServer/src/blazor-devserver.runtimeconfig.json (hunk 2 above)
Comment (captainsafia) [reply src/Components/WebAssembly/DevServer/src/blazor-devserver.runtimeconfig.json]: Updated this to use a template file.

File: src/Components/WebAssembly/DevServer/src/Microsoft.AspNetCore.Components.WebAssembly.DevServer.csproj [hunk 3]
```diff
@@ -19,6 +22,21 @@
     <Compile Include="$(SharedSourceRoot)CommandLineUtils\**\*.cs" />
   </ItemGroup>
 
+  <Target Name="_CreateRuntimeConfig" BeforeTargets="CoreBuild">
+    <PropertyGroup>
+      <_RuntimeConfigProperties>
+        MicrosoftAspNetCoreAppVersion=$(MicrosoftAspNetCoreAppVersion);
+      </_RuntimeConfigProperties>
+
+      <_RuntimeConfigPath>$(PublishDir)blazor-devserver.runtimeconfig.json</_RuntimeConfigPath>
```
Comment (pranavkm) [inline src/Components/WebAssembly/DevServer/src/Microsoft.AspNetCore.Components.WebAssembly.DevServer.csproj:31]:
```suggestion
      <_RuntimeConfigPath>$(IntermediateOutputPath)blazor-devserver.runtimeconfig.json</_RuntimeConfigPath>
```

perhaps?

**Review by HaoK — APPROVED** [review]

File: src/Components/WebAssembly/DevServer/src/blazor-devserver.runtimeconfig.json [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+    },
+    "rollForwardOnNoCandidateFx": 2
```
[the remark this answers is not in the document]
Comment (captainsafia) [reply src/Components/WebAssembly/DevServer/src/blazor-devserver.runtimeconfig.json]: It updates to the latest available framework version based on the major patch.

So if the version here says 6.0.0-preview.3 and the user has 6.0.0, 5.04, and 6.0.0-preview.3, it'll pick 6.0.0.

https://github.com/dotnet/core-setup/blob/92072fd9ea03740ea317b631d3894ff3c5d0854d/Documentation/design-docs/roll-forward-on-no-candidate-fx.md#roll-forward-in-absence-of-candidate-fx

File: eng/Versions.props [hunk 5]
```diff
@@ -261,6 +261,7 @@
     <XunitExtensibilityExecutionVersion>$(XunitVersion)</XunitExtensibilityExecutionVersion>
     <XUnitRunnerVisualStudioVersion>2.4.3</XUnitRunnerVisualStudioVersion>
     <MicrosoftDataSqlClientVersion>1.0.19249.1</MicrosoftDataSqlClientVersion>
+    <MicrosoftAspNetCoreAppVersion>6.0.0-preview.3.21167.1</MicrosoftAspNetCoreAppVersion>
```
Comment (pranavkm) [inline eng/Versions.props:264]: There's a `SharedFxVersion` variable that's configured at the repo level. (I think). You should use that

Comment (dougbu) [reply eng/Versions.props:264]: See https://github.com/dotnet/aspnetcore/blob/main/Directory.Build.targets#L47-L48. The variable is defined in every project that imports the Directory.Build.targets file. Note some projects are isolated i.e. don't import that file, some are test assets we want to be as realistic / normal as possible while other projects simply aren't .NET SDK projects.

File: src/Components/WebAssembly/DevServer/src/Microsoft.AspNetCore.Components.WebAssembly.DevServer.csproj [hunk 6]
```diff
@@ -19,6 +22,21 @@
     <Compile Include="$(SharedSourceRoot)CommandLineUtils\**\*.cs" />
   </ItemGroup>
 
+  <Target Name="_CreateRuntimeConfig" BeforeTargets="CoreBuild">
+    <PropertyGroup>
+      <_RuntimeConfigProperties>
+        MicrosoftAspNetCoreAppVersion=$(MicrosoftAspNetCoreAppVersion);
```
Comment (dougbu) [inline src/Components/WebAssembly/DevServer/src/Microsoft.AspNetCore.Components.WebAssembly.DevServer.csproj:28]:
```suggestion
        SharedFxVersion=$(SharedFxVersion);
```

File: src/Components/WebAssembly/DevServer/src/blazor-devserver.runtimeconfig.json.in [hunk 7]
```diff
@@ -0,0 +1,10 @@
+{
+  "runtimeOptions": {
+    "tfm": "net6.0",
+    "framework": {
+      "name": "Microsoft.AspNetCore.App",
+      "version": "${MicrosoftAspNetCoreAppVersion}"
```
Comment (dougbu) [inline src/Components/WebAssembly/DevServer/src/blazor-devserver.runtimeconfig.json.in:6]:
```suggestion
      "version": "${SharedFxVersion}"
```

**Review by dougbu — COMMENTED** [review]
I'm strongly in favour of using `$(SharedFxVersion)` because it'll stay up-to-date and set the minimum version used automatically. As-is, we could hit problems w/ users not getting the latest and greatest because we forget to update the property in eng/Versions.props right before a release.

Suggestions below&hellip;

**Review by pranavkm — APPROVED** [review]

**Pull request merged**